plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1884,21 +1886,21 @@ public Optional<Object> getInfo(ConnectorTableHandle table)
     public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         SchemaTableName schemaTableName;
+        boolean managed;
         String tableLocation;
         if (tableHandle instanceof CorruptedDeltaLakeTableHandle corruptedTableHandle) {
             schemaTableName = corruptedTableHandle.schemaTableName();
             tableLocation = corruptedTableHandle.location();
+            managed = corruptedTableHandle.managed();
         }
         else {
             DeltaLakeTableHandle handle = (DeltaLakeTableHandle) tableHandle;
             schemaTableName = handle.getSchemaTableName();
             tableLocation = handle.getLocation();
+            managed = handle.isManaged();
         }
 
-        Table table = metastore.getTable(schemaTableName.getSchemaName(), schemaTableName.getTableName())
```

Comment:
Add a test in `io.trino.plugin.deltalake.metastore.TestDeltaLakeMetastoreAccessOperations`

Reply:
good catch. the commit message was a lie, because there is transaction-level cache.

added a test and changed the message to
```
    Remove unnecessary cache reliance in Delta DROP TABLE

    The DROP TABLE flow looked like making redundant metastore calls,
    relying on the caching metastore being part of the transaction. This
    isn't needed and after the change it isn't there.
```